apps/server/src/modules/user/repo/role.repo.integration.spec.ts

```diff
@@ -0,0 +1,295 @@
+			expect(Object.keys(result).sort()).toEqual(
+				['createdAt', 'updatedAt', 'permissions', 'roles', 'name', '_id'].sort()
+			);
+		});
+
+		it('should return one role that matched by id', async () => {
+			const idA = new ObjectId().toHexString();
+			const idB = new ObjectId().toHexString();
+			const roleA = em.create(Role, { id: idA });
+			const roleB = em.create(Role, { id: idB });
+
+			await em.persistAndFlush([roleA, roleB]);
+			const result = await repo.findById(idA);
+			expect(result).toEqual(roleA);
+		});
+
+		it('should throw an error if roles by id not exist', async () => {
+			const idA = new ObjectId().toHexString();
+			const idB = new ObjectId().toHexString();
+			const roleA = em.create(Role, { id: idA });
+
+			await em.persistAndFlush([roleA]);
+			await expect(repo.findById(idB)).rejects.toThrow(NotFoundError);
+		});
+
+		it('should cache requested roles by id', async () => {
+			const idA = new ObjectId().toHexString();
+			const roleA = em.create(Role, { id: idA });
+
+			await em.persistAndFlush([roleA]);
+			expect(repo.cl1.get(idA)).toEqual(undefined);
+			await repo.findById(idA);
+			expect(repo.cl1.get(idA)).toEqual(roleA);
+		});
+
+		it('should select already cached roles instant of db call', async () => {
+			const idA = new ObjectId().toHexString();
+			const roleA = em.create(Role, { id: idA });
+			const spy = jest.spyOn(repo.cl1, 'get');
+
+			await em.persistAndFlush([roleA]);
+
+			expect(spy).not.toHaveBeenCalled();
+			await repo.findById(idA);
+			expect(spy).toHaveBeenCalled();
+			spy.mockRestore();
+		});
+	});
+
+	describe('resolvePermissionsById', () => {
+		afterEach(async () => {
+			repo.cl1.clear();
+			repo.cl2.clear();
+			await em.nativeDelete(Role, {});
+		});
+
+		it('should return right keys', async () => {
+			const idA = new ObjectId().toHexString();
+			const roleA = em.create(Role, { id: idA });
+
+			await em.persistAndFlush([roleA]);
+			const result = await repo.resolvePermissionsById(idA);
+			expect(Object.keys(result).sort()).toEqual(
+				['createdAt', 'updatedAt', 'permissions', 'roles', 'name', '_id'].sort()
+			);
+		});
+
+		it('should return one role that matched by name', async () => {
+			const idA = new ObjectId().toHexString();
+			const idB = new ObjectId().toHexString();
+			const roleA = em.create(Role, { id: idA });
+			const roleB = em.create(Role, { id: idB });
+
+			await em.persistAndFlush([roleA, roleB]);
+			const result = await repo.resolvePermissionsById(idA);
+			expect(result).toEqual(roleA);
+		});
+
+		it('should throw an error if roles by name not exist', async () => {
+			const idA = new ObjectId().toHexString();
+			const idB = new ObjectId().toHexString();
+			const roleA = em.create(Role, { id: idA });
+
+			await em.persistAndFlush([roleA]);
+			await expect(repo.resolvePermissionsById(idB)).rejects.toThrow(NotFoundError);
+		});
+
+		it('should resolve permissions of existing subroles', async () => {
```

Comment:
is this not happening on the other operations? If I use "FindById", do I only get the permissions of the main role?

Reply:
Yes findById put you the db result 1 to 1 without any operations. 
But after thinking about it ...the resolves are part of a role uc and not part of the repo. :D 🗡️